mpc_tools/mpc_phase2/mpc_client.cpp

```diff
@@ -28,10 +28,10 @@ void zeth_protoboard(libsnark::protoboard<Field> &pb)
 
 int main(int argc, char **argv)
 {
-    const std::map<std::string, subcommand *> commands{
+    const std::map<std::string, mpc_subcommand *> commands{
         {"phase2-contribute", mpc_phase2_contribute_cmd},
         {"phase2-verify-transcript", mpc_phase2_verify_transcript_cmd},
         {"create-keypair", mpc_create_keypair_cmd},
     };
-    return mpc_main(argc, argv, commands, zeth_protoboard);
+    return mpc_main(commands, zeth_protoboard, argc, argv);
```

Comment:
What's the rationale in the argument order switch here?

Reply:
This matches the order of the `libtool::run` call.  The first few parameters specialise the command, and the later args are per-invocation.  (You might think of the mpc_client `main` function as being the partial application of those first few arguments to `mpc_main`).

Reply:
I see, thanks.